modules/webgl/src/classes/program-configuration.js

```diff
@@ -9,6 +9,10 @@ export default class ProgramConfiguration {
     this.id = program.id;
     this.attributeInfos = [];
     this.attributeInfosByName = {};
+
+    // Locations may not be contiguous the case of matrix attributes
+    // so keep a separate location->attribute map.
+    this.attributeInfosByLocation = [];
```

Comment:
I can't see that we need both `attributeInfos` and `attributeInfosByLocation`. 

Fine with deleting either and just keeping one array that is sorted by real locations.

Nit: We may want to initialize the array to the MAX_ATTRIBUTES so we don't assign beyond array length.